x-pack/plugins/infra/server/routes/profiling/index.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { decodeOrThrow } from '@kbn/io-ts-utils';
+import { InfraProfilingRequestParamsRT } from '../../../common/http_api/profiling_api';
+import type { InfraBackendLibs } from '../../lib/infra_types';
+import { fetchProfilingFlamegraph } from './lib/fetch_profiling_flamechart';
+import { fetchProfilingStatus } from './lib/fetch_profiling_status';
+import { getProfilingDataAccess } from './lib/get_profiling_data_access';
+
+export function initProfilingRoutes({ framework, getStartServices, logger }: InfraBackendLibs) {
+  if (!Object.hasOwn(framework.plugins, 'profilingDataAccess')) {
+    logger.info(
+      "Skipping initialization of Profiling endpoints because 'profilingDataAccess' plugin is not available"
+    );
+    return;
+  }
+
+  framework.registerRoute(
+    {
+      method: 'get',
+      path: '/api/infra/profiling/status',
+      validate: false,
+    },
+    async (requestContext, request, response) => {
+      const coreRequestContext = await requestContext.core;
+      const infraRequestContext = await requestContext.infra;
+      const profilingDataAccess = await getProfilingDataAccess(getStartServices);
```

Comment:
You can perform all these actions in parallel. 
```suggestion
      cosnt [coreRequestContext, infraRequestContext, profilingDataAccess] = await Promise.all([
      requestContext.core,
      requestContext.infra,
      getProfilingDataAccess(getStartServices)
      ])
```

Reply:
True, thanks!